corehq/messaging/smsbackends/icds/models.py

```diff
@@ -0,0 +1,129 @@
+from urllib import urlencode
+from urllib2 import urlopen
+
+from crispy_forms import layout as crispy
+from django.conf import settings
+from django.utils.translation import ugettext_lazy as _
+
+from corehq.apps.sms.mixin import BackendProcessingException
+from corehq.apps.sms.models import SQLSMSBackend
+from corehq.apps.sms.forms import BackendForm
+from dimagi.utils.django.fields import TrimmedCharField
+
+ERROR_CODES = {
+    "-2": "Invalid credentials",
+    "-3": "Empty mobile number",
+    "-4": "Empty message",
+    "-5": "HTTPS disabled",
+    "-6": "HTTP disabled",
+    "-13": "Internal Error",
+    "-201": "Email Delivery Disabled",
+    "-401": "Invalid Scheduled Time",
+    "-404": "Invalid MsgType",
+    "-406": "Invalid Port",
+    "-407": "Invalid Expiry minutes",
+    "-408": "Invalid Customer Reference Id",
+    "-409": "Invalid Bill Reference Id",
+    "-410": "Invalid Destination Address",
+    "-432": "Invalid Bill Reference Id Length",
+    "-433": "Invalid Customer Reference Id Length",
+}
+
+RETRY_ERROR_CODES = {"-13"}
```

Comment:
I think that this is the only code that we can retry on and expect a different response. everything else looks like user error except -5 and -6